### PR TITLE
Items browse prefix filter

### LINKED ISSUE #8203: Items browse not filtered, jump to functionality causes issues
**Describe the bug**
The items browse, eg [/browse/title?startsWith=T](https://demo7.dspace.org/browse/title?startsWith=T), does not filter the results with the "startsWith" but rather jumps to the page where items starting with this query are.

This is however in opposition, and causing the same issues, that lead to the changes done here: https://github.com/DSpace/DSpace/pull/1730

The plain-text metadata browse does have entries that get filtered by the startsWith query, eg /browse/subject?startsWith=T

**To Reproduce**
Items browse
1. Go to items browse [/browse/title](https://demo7.dspace.org/browse/title)
2. Add a startsWith query/letter, for which there are no results,  [/browse/title?startsWith=X](https://demo7.dspace.org/browse/title?startsWith=X)
3. Looking at results from REST https://api7.dspace.org/server/#/server/api/discover/browses/title/items?sort=dc.title,ASC&page=0&size=20&startsWith=X => Still shows results, not filtered

Plain-text metadata browse:
1. Go to metadata browse [/browse/subject](https://demo7.dspace.org/browse/subject)
2. Add a startsWith query/letter, for which there are no results,  eg [/browse/author?startsWith=stent](https://demo7.dspace.org/browse/author?startsWith=stent)
3. Looking at results from REST https://api7.dspace.org/server/#/server/api/discover/browses/subject/entries?sort=default,ASC&page=0&size=20&startsWith=stent => No results, since filtered

Items browse not all browse results completely pageable with jump to method either:
- [browse/title?startsWith=14](https://api7.dspace.org/server/#/server/api/discover/browses/title/items?sort=dc.title,ASC&page=0&size=20&startsWith=14) vs [/browse/title](https://api7.dspace.org/server/#/server/api/discover/browses/title/items?sort=dc.title,ASC&page=0&size=20)
- On https://demo7.dspace.org/browse/title?startsWith=14&bbm.rpp=100 => Can't navigate back to previous page, even though there are items that alphabetically precede it, ex "090952/Z/09/Z" item in complete title browse

**Expected behavior**
Both types of browses should behave similarly on whether they filter their results or jump to the page with expected results.
The jump-to method however causes unexpected results for the items browse and is likely not possible for the metadata browse (since it works with facetting), so both should act as a filter.

**Related work**
- Similarly fixed before here: https://github.com/DSpace/DSpace/pull/1730
- Both items and metadata values browse can be clearly labeled as filtering results after this is done, like suggested in: https://github.com/DSpace/dspace-angular/issues/1379

### PR DESCRIPTION
**Pull request by MarieVerdonck**
## References
* Fixes https://github.com/DSpace/DSpace/issues/8203
* UI clarifications on filter functionality (now applied to both plain text and items browsing) will be addressed in PR of https://github.com/DSpace/dspace-angular/issues/1379
* Related to `/api/discover/browses/<:index-name>/items` [REST Contract](https://github.com/DSpace/RestContract/blob/main/browses.md), **needs to be altered** there as well, still states _"startsWith: the value to use to calculate the offset. Each value in the page will be greater or equals than the specified value"_

## Description
The 'items' browse (`/api/discover/browses/<:index-name>/items`) now also filters the results when a `startsWith` is added, rather than jumping to the page where items with that prefix (in `dc.title`) are located.

## Instructions for Reviewers
The browse by items with `startsWith`, e.g. [/browse/title?startsWith=T](https://demo7.dspace.org/browse/title?startsWith=T) => `/server/api/discover/browses/title/items?sort=dc.title,ASC&page=0&size=20&startsWith=T` should now filter its results:
- The results should only contain items (paginated) that have title starting with 'T'
- Results should start at first page (totalElement less than without startsWith because now filtered)

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java [hunk 1]
```diff
@@ -271,7 +274,7 @@ private BrowseInfo browseByItem(BrowserScope bs)
                 log.debug("browsing using focus: " + focusValue);
 
                 // Convert the focus value into an offset
-                offset = getOffsetForValue(focusValue);
+                // offset = getOffsetForValue(focusValue);
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java:277]: Should this commented out code be removed?

File: dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java [hunk 2]
```diff
@@ -684,13 +687,13 @@ private int getTotalResults(boolean distinct)
         // our count, storing them locally to reinstate later
         String focusField = dao.getJumpToField();
         String focusValue = dao.getJumpToValue();
-        String orderField = dao.getOrderField();
+        // String orderField = dao.getOrderField();
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/browse/BrowseEngine.java:690]: Again, it looks like you've commented out code here, and on line 696 and 706.  Should these be removed?  It looks (to me) like the ordering may not be necessary in this `getTotalResults()` method, as it doesn't matter what order the results come back in.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@MarieVerdonck and @antoine-atmire :  Overall, this code looks good & it works for me.  However, this PR currently includes some commented out code...and that should be cleaned up before we can merge it.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @MarieVerdonck and @antoine-atmire !  This now looks good to me.  Could one of you submit a PR to update the REST Contract (as noted in the description)?  Once that REST Contract PR is in place, I'll get this merged along with the REST Contract updates.

**Comment by MarieVerdonck** [comment]
@tdonohue : PR to update rest contract to reflect these changes: https://github.com/DSpace/RestContract/pull/184 
startsWith is now filter for both item and plain-text browses (previously only for plain-text)

**Comment by tdonohue** [comment]
Merging, as I just reviewed the REST Contract PR and also merged it https://github.com/DSpace/RestContract/pull/184  Thanks again @MarieVerdonck and @antoine-atmire !

**Pull request merged**